pkg/analyzer/analyzers/github/finegrained/finegrained.go

```diff
@@ -1336,31 +1331,32 @@ func PrintFineGrainedToken(cfg *config.Config, info *common.SecretInfo) {
 	common.PrintGists(info.Gists, cfg.ShowAll)
 }
 
-func printFineGrainedPermissions(accessMap map[string]Permission, showAll bool, repoPermissions bool) {
+func printFineGrainedPermissions(accessMap []Permission, showAll bool, repoPermissions bool) {
 	permissionCount := 0
 	t := table.NewWriter()
 	t.SetOutputMirror(os.Stdout)
 	t.AppendHeader(table.Row{"Permission Type", "Permission" /* Add more column headers if needed */})
 
 	// Extract keys from accessMap into slice
 	keys := make([]string, 0, len(accessMap))
-	for k := range accessMap {
-		keys = append(keys, k)
+	for _, k := range accessMap {
+		val, _ := k.ToString()
+		keys = append(keys, val)
 	}
 	// Sort the slice
 	sort.Strings(keys)
 
-	for _, key := range keys {
-		value := accessMap[key]
-		if value == Invalid {
+	for idx, permStr := range keys {
+		perm := accessMap[idx]
```

Comment:
If we're relying on the index on line 1350, I think we also need to remove the `sort` on 1347.

Reply:
Fixed, just accessing directly now